dds/DCPS/DomainParticipantFactoryImpl.cpp

```diff
@@ -100,7 +100,7 @@ DomainParticipantFactoryImpl::create_participant(
   // needed for the built in topics during enable.
   dp->set_object_reference(dp_obj);  //xxx no change
 
-  if (qos_.entity_factory.autoenable_created_entities == 1) {
+  if (qos_.entity_factory.autoenable_created_entities == true) {
```

Comment:
I am 100% in favor of improving the code in general, looks with this change I focused to much on consistent style instead of general improvements. I will fix the comparison of autoenable_created_entities right now, give me a few minutes